packages/workflows/src/definition/product/prepare-create-inventory-items.ts

```diff
@@ -0,0 +1,47 @@
+import { ProductTypes } from "@medusajs/types"
+import { WorkflowArguments } from "../../helper"
+
+type AssociationTaggedVariant = ProductTypes.ProductVariantDTO & {
+  _associationTag?: string
+}
+
+export async function prepareCreateInventoryItems({
+  data,
+}: WorkflowArguments<{
+  products: ProductTypes.ProductDTO[]
```

Comment:
todo: I believe we should expect an array of object that contains variants and not necessarily an entire product here to make it more reusable elsewhere. wdyt?